File: estimator/turbofan/src/main/kotlin/com/alessandropacielli/turbofan/data/Repository.java
package com.alessandropacielli.turbofan.data;

import java.util.List;

public interface Repository<Model> {
    /**
     * Returns the last n measurements for the required device
     *  @param device : String, identifies the device
     * @param n :      Int, limits the query to the needed amount of data points
     * @return
     */
    public List<Model> getLastMeasurements(String device, int n);
}

File: estimator/turbofan/src/main/kotlin/com/alessandropacielli/turbofan/data/Repository.kt
package com.alessandropacielli.turbofan.data

interface Repository<Model> {
    /**
     * Returns the last n measurements for the required device
     * @param device: String, identifies the device
     * @param n: Int, limits the query to the needed amount of data points
     */
    fun getLastMeasurements(device: String, n: Int): MutableList<Model>
}